src/main/java/org/prebid/server/auction/CurrencyService.java

```diff
@@ -0,0 +1,128 @@
+package org.prebid.server.auction;
+
+import org.apache.commons.collections4.MapUtils;
+import org.prebid.server.auction.model.Currency;
+import org.prebid.server.exception.PreBidException;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Service for price currency conversion between currencies.
+ */
+public class CurrencyService {
+
+    private static final Integer SCALE = 5;
```

Comment:
I am not sure about scale after currency conversion. Rates from currency server have 4 numbers after point, also they have rates like "USD" to "IDR":13775, so 1 IDR is equal to 0,000072595. If scale will be less than 5, IDR will converted to 0, which can cause issues later, because in response validator we drop all prices less than 0 and after assume, that 0 is not possible value for bid price

Reply:
There is a suggestion to use BigDecimal#precision() instead of hardcoded scale value. This way reverse conversion of significant currency rate's like USD : IDR  will compute a value which gives quite accurate results :  
13775 reverse conversion output is 0.00007
130 - 0.008
600000 -0.000002
1.2 - 0.83